oneflow/api/python/autograd/autograd.cpp

```diff
@@ -18,23 +18,45 @@ limitations under the License.
 #include <memory>
 #include <vector>
 #include "oneflow/api/python/of_api_registry.h"
+#include "oneflow/core/framework/tensor.h"
 #include "oneflow/core/framework/tensor_tuple.h"
+#include "oneflow/core/framework/op_expr_helper.h"
 #include "oneflow/core/autograd/autograd_engine.h"
+#include "oneflow/core/framework/op_interpreter_util.h"
 #include "oneflow/core/common/util.h"
 
 namespace oneflow {
 namespace autograd {
 
 namespace {
 
+bool IsScalarTensor(const one::Tensor& tensor) {
+  const auto& shape = tensor.shape();
+  return shape->elem_cnt() == 1;
+}
+
 // Checks and sets default value for initial gradients based on out_grads
 // If output is the tensor whose size is greater than 1, out_grad's shape must be same as output's.
-// If output is a scaler tensor, out_grad will also be a scaler or empty(will be inited to
+// If output is a scalar tensor, out_grad will also be a scaler or empty(will be inited to
 // `flow.ones([1])`).
 Maybe<one::TensorTuple> CheckAndInitOutGrads(const one::TensorTuple& outputs,
                                              const one::TensorTuple& out_grads) {
   auto gradients = std::make_shared<one::TensorTuple>(out_grads.size());
-  TODO();  // Wang Yinggang. Checks all out_grads and push default value for empty item
+  for (int i = 0; i < out_grads.size(); ++i) {
+    if (out_grads.at(i).get()) {
+      CHECK_OR_RETURN(*(outputs.at(i)->shape()) == *(out_grads.at(i)->shape()))
+          << "out_grad's shape must be same as output's";
```

Comment:
<< "The out_grad shape  must be same as output. (" << outputs.at(i)->shape()->ToString() << " vs " << out_grads.at(i)->shape()-> ToString() << ")";